### DISCUSSION TITLE
how to copy files to .output/public using nitro publicAssets?

### DISCUSSION THREAD

**Discussion by lucassonsinlima** [nuxt/nuxt#20333] [Questions] [answered]
Hi! 

When using `npm run generate`, I'm trying to use some font files from another library, so I tried to do something like this:

based on this nitro [docs](https://nitro.unjs.io/config#publicassets) section

```
...
nitro: {
  publicAssets: [
      {
        baseURL: 'node_modules/xyz/fonts',
        dir: "public/fonts"
      }
  ]
}
```

I saw some related issues and seems that this is the recommended way to achieve this behavior but doesn't works. Maybe I misunderstood how it should works.

I've created a repository showing a simpler example of what I need, using a root `images` path as assets origin.

https://github.com/lucassonsinlima/reproduce-nitro-public-assets

Related issued and discussions:
https://github.com/nuxt/nuxt/issues/18918
https://github.com/nuxt/nuxt/discussions/17691
https://github.com/nuxt/nuxt/issues/15779#issuecomment-1397418353

**Comment by lucassonsinlima** [comment, marked as answer]
after some tests I achieved the expected behavior using `nitro:config` hook. 

```
hooks: {
  'nitro:config' (nitroConfig) {
    const resolver = createResolver(import.meta.url)
    const filePath = resolver.resolve(nitroConfig.rootDir, 'node_modules/xyz/fonts')
    
    nitroConfig.publicAssets.push({
      baseURL: '/fonts',
      dir: filePath
    })
  }
},
```

**Comment by gleb-svechnikov** [comment]
Not sure about nitro:config hook, never used it. But I've figured out how to use _publicAssets_
Problem is in relative path. I've found [nuxt project](https://github.com/i7en0v91/travel-agency-website/blob/027b951449021290ce26423b423965f23d686ec9/src/nuxt.config.ts ) where this approach was working and noticed that _dir_ has ```./../``` in it. So in your case it should be 
``` 
...
nitro: {
  publicAssets: [
      {
        baseURL: '/fonts',
        dir: "./../node_modules/xyz/fonts"
      }
  ]
}
```
